engine/src/lib.rs

```diff
@@ -481,6 +555,30 @@ mod contract {
         io.return_output(block_hash.as_bytes())
     }
 
+    #[no_mangle]
+    pub extern "C" fn get_previous_block_hashchain() {
```

Comment:
In this method, I'm returning the previous block hashchain, which is the hashchain of the last completed block. Do you think it will be better to return a tuple` <block_height, block_hashchain>`, so it is clear to which block belongs?

As a note for your consideration about this, near skip blocks and near blocks without aurora txs, are accounted on the hashchain computation. This means that if we have a block H with some aurora txs, then a block H + 1 as a skip block or a block without aurora txs, and then at block H + 2 we call `get_previous_block_hashchain`, we will get the hashchain of H + 1 and not the one of block H.

Reply:
Yes, good idea. I agree it should return the block height associated with the hash, for clarity.